src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4400,350 +4405,662 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
+                    var resultType = isContainer
+                        ? CompletionResultType.ProviderContainer
+                        : CompletionResultType.ProviderItem;
+                    
+                    bool leafQuotesNeeded;
+                    var completionText = NewPathCompletionText(
+                        basePath,
+                        EscapePath(entryName, stringType, literalPaths, out leafQuotesNeeded),
+                        stringType,
+                        containsNestedExpressions: false,
+                        forceQuotes: baseQuotesNeeded || leafQuotesNeeded,
+                        addAmpersand: false);
+                    results.Add(new CompletionResult(completionText, entryName, resultType, entry.FullName));
+                }
+            }
 
-                        if (psobjs.Count > 0 && !LocationGlobber.StringContainsGlobCharacters(wordToComplete))
-                        {
-                            string leaf = null;
-                            string pathWithoutProvider = wordContainsProviderId
-                                    ? wordToComplete.Substring(wordToComplete.IndexOf(':') + 2)
-                                    : wordToComplete;
+            return results;
+        }
 
-                            try
-                            {
-                                leaf = Path.GetFileName(pathWithoutProvider);
-                            }
-                            catch (Exception)
-                            {
-                            }
+        /// <summary>
+        /// Helper method for generating path completion results standard providers that don't need any special treatment.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="provider"></param>
+        /// <param name="resolvedPaths"></param>
+        /// <param name="filterText"></param>
+        /// <param name="containersOnly"></param>
+        /// <param name="relativePaths"></param>
+        /// <param name="literalPaths"></param>
+        /// <param name="inputUsedHome"></param>
+        /// <param name="providerPrefix"></param>
+        /// <param name="stringType"></param>
+        /// <returns></returns>
+        private static List<CompletionResult> GetDefaultProviderResults(
+            CompletionContext context,
+            ProviderInfo provider,
+            Collection<PSObject> resolvedPaths,
+            string filterText,
+            bool containersOnly,
+            bool relativePaths,
+            bool literalPaths,
+            bool inputUsedHome,
+            string providerPrefix,
+            StringConstantType stringType)
+        {
+            string homePath;
+            if (inputUsedHome && !string.IsNullOrEmpty(provider.Home))
+            {
+                homePath = provider.Home;
+            }
+            else
+            {
+                homePath = null;
+            }
```

Comment:
```suggestion
            string homePath = null;
            if (inputUsedHome && !string.IsNullOrEmpty(provider.Home))
            {
                homePath = provider.Home;
            }
```

Reply:
While it obviously won't make a noticeable difference, won't that be slightly less efficient than the current if/else approach?

Reply:
It is only style comment (to reduce number of code lines and make the code more readable).

Reply:
In that case I will use a ternary expression instead.